python-lib/symspell_checker.py

```diff
@@ -0,0 +1,437 @@
+            language: Language code in ISO 639-1 format
+
+        Returns:
+            List of tuples with 3 elements:
+                1. Corrected spaCy document
+                2. List of misspellings as strings
+                3. Number of misspellings
+        """
+        start = time()
+        logging.info("Spellchecking {:d} documents in language '{}'...".format(len(document_list), language))
+        tuple_list = [("", [], 0)] * len(document_list)
+        try:
+            self._add_symspell_checker(language)
+            doc_lang_iterator = ((doc, language) for doc in document_list)
+            with ThreadPoolExecutor(max_workers=self.DEFAULT_NUM_THREADS) as executor:
+                tuple_list = list(executor.map(lambda x: self.check_document(*x), doc_lang_iterator))
+            logging.info(
+                "Spellchecking {:d} documents in language '{}': Done in {:.2f} seconds.".format(
+                    len(tuple_list), language, time() - start
+                )
+            )
+        except ValueError as e:
+            logging.warning(
+                "Spellchecking error: '{}' for documents: '{}' in language '{}', output columns will be empty".format(
+                    e, truncate_text_list([d.text for d in document_list]), language
+                )
+            )
+        return tuple_list
+
+    def _prepare_df_for_spellchecker(
+        self, df: pd.DataFrame, text_column: AnyStr, language_column: AnyStr, language: AnyStr
+    ) -> None:
+        """Private method to prepare a Pandas dataframe in-place before feeding it to the spellchecker
+
+        Tokenizes the content of the text column into a new column containing spaCy documents
+        Adds new columns to hold the future outputs of the spellchecker
+
+        Args:
+            df: Input pandas DataFrame
+            text_column: Name of the column containing text data
+            language_column: Name of the column with language codes in ISO 639-1 format
+            language: Language code in ISO 639-1 format
+                If equal to "language_column" this parameter is ignored in favor of language_column
+        """
+        self._output_column_description_dict = OrderedDict()
+        for k, v in self.OUTPUT_COLUMN_DESCRIPTION_DICT.items():
+            column_name = generate_unique(k, df.keys(), text_column)
+            df[column_name] = pd.Series([""] * len(df.index))
+            self._output_column_description_dict[column_name] = v
+        self._tokenizer.tokenize_df(df, text_column, language_column, language)
+
+    def _format_output_df(self, df: pd.DataFrame) -> None:
+        """Private method to format the output dataframe after spellchecking
+
+        Removes the tokenized column with spaCy documents
+        Replaces empty lists of misspellings by an empty string
+
+        Args:
+            df: Input pandas DataFrame
+        """
+        del df[self._tokenizer.tokenized_column]
```

Comment:
it seems to be the recommended way of removing columns in pandas
```suggestion
       df.drop(self._tokenizer.tokenized_column, axis=1, inplace=True)
```